mwclient/client.py

```diff
@@ -433,7 +469,16 @@ def _query_string(*args, **kwargs):
         ]
         return OrderedDict(qs1 + qs2)
 
-    def raw_call(self, script, data, files=None, retry_on_error=True, http_method='POST'):
+    def raw_call(
+        self,
+        script: str,
+        data: Mapping[str, Any],
+        files: Optional[
+            Mapping[str, Union[io.BytesIO, Tuple[str, io.BufferedReader]]]
```

Comment:
I think this is a bit too restrictive in various ways, see https://requests.readthedocs.io/en/latest/user/quickstart/#post-a-multipart-encoded-file . It seems like a `Tuple[str, str]` is also valid, and three- or four-item tuples are also possible (to specify content type and headers). And, uh, `files` can also apparently be a list of tuples - https://requests.readthedocs.io/en/latest/user/advanced/#post-multiple-multipart-encoded-files . Whee, isn't typing fun

Reply:
Yeah, the actual type (according to `types-requests` is something like the following:

```python
files: Mapping[str, SupportsRead[str | bytes] | str | bytes | tuple[str | None, SupportsRead[str | bytes] | str | bytes] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, Mapping[str, str]]] | Iterable[tuple[str, SupportsRead[str | bytes] | str | bytes | tuple[str | None, SupportsRead[str | bytes] | str | bytes] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, Mapping[str, str]]]] | None = None
```

I think we just have to come up with a reasonable subset here to be honest 😅

Reply:
yeah, fair enough.